### PR TITLE
Merge the many-framework into this repository

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -70,6 +72,58 @@ jobs:
...
+      - checkout
+      - restore_cache:
+          keys:
+            - cargo-build-{{ .Environment.MANY_RS_CACHE_VERSION }}-{{ arch }}-{{ checksum "Cargo.lock" }}
+      - rust/install:
+          version: nightly
+      # TODO: Make a release with `many` CLI and install the release instead
+      - run:
+          name: install `many` CLI
+          command: cargo install --force --git https://github.com/liftedinit/many-rs many
```
Comment (fmorency) [inline .circleci/config.yml:90]: You can use `--path` here now

File: .circleci/config.yml [hunk 2]
```diff
@@ -167,11 +223,112 @@ jobs:
...
+    executor: linux2204
+    resource_class: large
+    steps:
+      - checkout
+      # TODO: Make a release with `many` CLI and install the release instead
+      - rust/install:
+          version: nightly
+      - run:
+          name: install `many` CLI
+          command: cargo install --git https://github.com/liftedinit/many-rs many
```
Comment (fmorency) [inline .circleci/config.yml:290]: Use `--path` here too

File: .circleci/config.yml [hunk 3]
```diff
@@ -207,6 +364,18 @@ workflows:
           matrix:
             parameters:
               os: [linux2204, macos]
```
Comment (fmorency) [inline .circleci/config.yml:366]: Can we leave `macos` for nightly?

File: .circleci/config.yml [hunk 4]
```diff
@@ -70,6 +72,58 @@ jobs:
...
+      - detect/init
+      - rust/install:
+          version: nightly
+      - rust/build:
+          release: true
+          with_cache: false
+      - run: mkdir -p artifacts
+      - run:
+          name: creating release archive
+          command: file target/release/* | grep 'executable\|shared object' | cut -d ':' -f 1 | xargs tar czvf artifacts/many-rs_PR_${CIRCLE_PR_NUMBER}_${CIRCLE_SHA1}_${OSD_ID}_${OSD_VERSION}.tar.gz
```
Comment (fmorency) [inline .circleci/config.yml:121]: nit: For some reason, we have some `.so` in the archive now.

**Review by fmorency — CHANGES_REQUESTED** [review]
First series of comments. I'm currently testing this PR.

File: .circleci/config.yml (hunk 1 above)
Comment (hansl) [reply .circleci/config.yml:90]: Separate PR. I'll do a follow up.

File: .circleci/config.yml (hunk 2 above)
Comment (hansl) [reply .circleci/config.yml:290]: Separate PR. I'll do a follow up.

File: .circleci/config.yml (hunk 3 above)
Comment (hansl) [reply .circleci/config.yml:366]: Separate PR. I'll do a follow up.

File: .circleci/config.yml (hunk 4 above)
Comment (hansl) [reply .circleci/config.yml:121]: Let me check.

**Comment by hansl** [comment]
I also forgot to update the README. BRB

**Comment by hansl** [comment]
No idea why we get `.so` files in the archive. Let's discuss in our 1:1

**Comment by hansl** [comment]
Also, where's the codecov report? mmmmh

**Review by fmorency — APPROVED** [review]
I tested
- cargo build
- cargo test
- make code-coverage
- bazel build
- bazel test
- nix docker build

And everything seems to work. Please fix the remaining comments/issues in other PRs. Thanks!

Bazel code coverage failed, but I pushed the fix on this PR.

File: README.md [hunk 5]
```diff
@@ -36,22 +240,77 @@ The dependency graph between the crates in this repository looks like this:
 
 ```mermaid
 graph TD;
+  many-client --> http_proxy;
```
[the remark this answers is not in the document]
Comment (hansl) [reply README.md:243]: Maybe not :/

**Review by MavenRain — APPROVED** [review]

**Pull request merged**